src/components/CartList/CartList.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { useContext, useEffect, useState } from "react";
+import { CartContext } from "../../contexts";
+import { Meals } from "../../models";
+import { ItemList } from "../../UI";
+
+export const CartList = () => {
+    const cartCtx = useContext(CartContext);
+    const allItems: Meals[] = cartCtx.items;
+    const [totalAmount, setTotalAmount] = useState(0);
+
+
+    const addItem = (item: Meals) => {
+        cartCtx.addItem({ ...item, quantity: 1 });
+    }
+
+    const removeItem = (id: string) => {
+        cartCtx.removeItem(id);
+    }
+
+    useEffect(() => {
+        if (allItems.length > 0) {
+            const amount: number = allItems.reduce((current, item) => {
+                return current + item.price * item.quantity;
+            }, 0)
+            setTotalAmount(amount);
+        } else {
+            setTotalAmount(0);
+        }
+    }, [allItems])
```

Comment:
Do you really need a `useEffect` here? `addItem` and `removeItem` should simply change the context and the component will be redrawn anyway. Let the component render by itself and the `useEffect` logic in the body of the component. This is not a good use case for `useEffect`.